yarn-project/noir-contracts/src/contracts/non_native_token_contract/src/main.nr

```diff
@@ -200,36 +215,42 @@ contract NonNativeToken {
         let mut initialContext = PrivateFunctionContext::new();
         initialContext.args = initialContext.args.push_array([amount, sender.x, sender.y, recipient.x, recipient.y]);
 
+        // Gets the set of sender's notes and picks 2 of those.
```

Comment:
why 2 btw? (for my understanding?)

Reply:
I think the amount of notes is fixed because of some limitation of Noir and this was the simplest way to get something to work. But this just my guess. I will forward this to noir-questions.